demos/https/iot_demo_https_common.c

```diff
@@ -0,0 +1,227 @@
+{
+    /* Status of HTTPS Client API. */
+    IotHttpsReturnCode_t httpsClientStatus = IOT_HTTPS_OK;
+    /* The HTTPS request configurations for getting the file size. */
+    IotHttpsRequestInfo_t fileSizeReqConfig = { 0 };
+    /* The HTTPS response configurations for getting the file size. */
+    IotHttpsResponseInfo_t fileSizeRespConfig = { 0 };
+    /* Synchronous request specific configurations. */
+    IotHttpsSyncInfo_t reqSyncInfo = { 0 };
+    /* Synchronous response specific configurations. */
+    IotHttpsSyncInfo_t respSyncInfo = { 0 };
+
+    /* Handle identifying the HTTP request. This is valid after the request has been initialized with
+     * IotHttpsClient_InitializeRequest(). */
+    IotHttpsRequestHandle_t fileSizeReqHandle = NULL;
+
+    /* Handle identifying the HTTP response. This is valid after the reponse has been received with
+     * IotHttpsClient_SendSync(). */
+    IotHttpsResponseHandle_t fileSizeRespHandle = NULL;
+
+    /* The status of HTTP response for this request. */
+    uint16_t respStatus = IOT_HTTPS_STATUS_OK;
+
+
+    /* String to store the Content-Range header field value. This header field as we are requesting in this demo is of
+     * the form: "Content-Range: bytes 0-0/FILESIZE", where file size would be the length of the maximum 32 bit integer
+     * which is 10.  Since the header field value "bytes 0-0/FILESIZE" is less than the maximum possible Range header
+     * field value, we size this string to the Range header field value.*/
+    char contentRangeValStr[ RANGE_VALUE_MAX_LENGTH ] = { 0 };
+    /* The location of the file size in the contentRangeValStr. */
+    char * pFileSizeStr = NULL;
+    /* Size in bytes of a single character. */
+    uint8_t sizeOfOneChar = 1;
+
+    /* We are retrieving the file size synchronously because we cannot run this demo without the file size anyways
+     * so it's OK to block. */
+    reqSyncInfo.pBody = NULL;  /* This is a GET request so there is no data in the body. */
+    reqSyncInfo.bodyLen = 0;   /* Since there is not data in the body the length is 0. */
+    respSyncInfo.pBody = NULL; /* We don't care about the body in this request since we want the file size that will
+                                * be found in a header value. */
+    respSyncInfo.bodyLen = 0;  /* Since we are not saving any response body the length is 0. */
+
+    /* Set the request configurations. */
+    fileSizeReqConfig.pPath = pPath;
+    fileSizeReqConfig.pathLen = pathLen;
+    fileSizeReqConfig.pHost = pAddress;
+    fileSizeReqConfig.hostLen = addressLen;
+    fileSizeReqConfig.method = IOT_HTTPS_METHOD_GET; /* Performing a HEAD request with S3 requires generating a Sigv4 signature
+                                                      * in an Authorization header field. We work around this by performing a GET
+                                                      * on Range: bytes=0-0. Then extracting the size of the file from the
```

Comment:
Could these last two sentences be clarified?

Reply:
/* Performing a HEAD request with S3 requires generating a Sigv4 signature
                                                      * in an Authorization header field. We work around this by performing a GET
                                                      * on Range: bytes=0-0. The size of the file is extracted from the Content-Range 
                                                      * header field in the response. */